### PR TITLE
focus the terminal when a letter key is pressed in the accessible view

### LINKED ISSUE #195582: [Accessibility] Support a custom setting for users to jump to terminal input from accessible buffer by typing
Type: <b>Feature Request</b>

CC @meganrogge 

I am a heavy terminal user and very frequently go back and forth accessible terminal view and terminal input.

In order to type something after verifying the terminal output in the buffer, I have to either press ctrl+DownArrow or ESC key first.

It would be even better if I could start typing in terminal buffer and it directly goes to the terminal input. Plese consider adding this c customization for some advanced users for their improved productivity.

VS Code version: Code - Insiders 1.84.0-insider (4678919d52909efc7c4ab958a2e5c8f45d50ffc3, 2023-10-13T05:34:13.175Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Remote OS version: Linux x64 4.18.0-477.27.1.el8_8.x86_64

### PR DESCRIPTION
**Pull request by meganrogge**
fix #195582

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 1]
```diff
@@ -908,92 +908,8 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
...
-			// If tab focus mode is on, tab is not passed to the terminal
-			if (TabFocus.getTabFocusMode() && event.key === 'Tab') {
-				return false;
-			}
-
-			// Prevent default when shift+tab is being sent to the terminal to avoid it bubbling up
-			// and changing focus https://github.com/microsoft/vscode/issues/188329
-			if (event.key === 'Tab' && event.shiftKey) {
-				event.preventDefault();
-				return true;
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:980]: only change to this whole thing was removing this

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:980]: which you can verify with the cool moved code feature from @hediet

Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:980]: Wasn't this needed to support tab+shift being passed through to the shell?

File: src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBufferProvider.ts [hunk 2]
```diff
@@ -50,6 +51,13 @@ export class TerminalAccessibleBufferProvider extends DisposableStore implements
 		}));
 	}
 
+	onKeyDown(e: IKeyboardEvent): void {
+		if (!this._instance.shouldProcessKeyEvent(e.browserEvent) || !isSingleLetterKey(e.browserEvent)) {
+			return;
+		}
+		this._instance.focus();
+	}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBufferProvider.ts:59]: Does this work by `keyup` getting handled by xterm.js?

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 3]
```diff
@@ -1053,6 +969,86 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
...
+		const EXCLUDED_KEYS = ['RightArrow', 'LeftArrow', 'UpArrow', 'DownArrow', 'Space', 'Meta', 'Control', 'Shift', 'Alt', '', 'Delete', 'Backspace', 'Tab'];
+
+		// only keep track of input if prompt hasn't already been shown
+		if (this._storageService.getBoolean(SHOW_TERMINAL_CONFIG_PROMPT_KEY, StorageScope.APPLICATION, true) &&
+			!EXCLUDED_KEYS.includes(event.key) &&
+			!event.ctrlKey &&
+			!event.shiftKey &&
+			!event.altKey) {
+			this._hasHadInput = true;
+		}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:1000]: This has a bunch of side effects so exposing it, especially with the name `shouldProcessKeyEvent` is misleading. I'm also a little worried about exposing such an internal function on `ITerminalInstance`. Is `isSingleLetterKey` good enough to catch most cases?

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts (hunk 3 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:1000]: Hmm, I will explore some more. I was mostly concerned with ensuring the accessible view isn't exited inappropriately. It seemed like I'd be duplicating a lot of this code, which is why I exposed it.

File: src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBufferProvider.ts (hunk 2 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBufferProvider.ts:59]: Yes

**Review by Tyriar — APPROVED** [review]

**Pull request merged**